### ISSUE TITLE
L-functions for Galois-conjugate HMFs

### ISSUE THREAD

**Issue by alinabucur** [LMFDB/lmfdb#3328]
The L-functions at 

http://www.lmfdb.org/ModularForm/GL2/TotallyReal/2.2.5.1/holomorphic/2.2.5.1-121.3-a

and 

http://www.lmfdb.org/L/ModularForm/GL2/TotallyReal/2.2.5.1/holomorphic/2.2.5.1-121.1-b

are the same, but they don't seem to know about each other or the other modular form they come from. First of all, I don't know enough about Hilbert modular forms to know if the fact that the L-functions of the two forms are the same is a mistake (I expected these two L-functions to be Galois-conjugate, but not the same, since the forms are like that), so an expert should check that. If it is correct, the L-functions should know about each other and point to one another in the related objects. Even if they turn out not to be the same, maybe they need to point to each other anyway.

**Comment by AndrewVSutherland** [comment]
Part of the plan for Release 1.2 is to have precomputed L-functions in the database in a standard form where they can be easily (and under suitable hypotheses, even rigorously) compared, for all of the arithmetic objects where this can be reasonably done up to some degree/weight/conductor bound (which would certainly include all the HMFs).  Once this is done we would like to make L-function friends completely automatic, so that objects in the LMFDB will automatically have friend links to all other objects in the LMFDB with the same L-function.

We are not quite there yet, but substantial progress was made over the summer and we know have nearly all the tools in place to do this.  It's now just a matter of getting it done, but is something Edgar Costa, Andy Booker, John Jones, myself and other plan to focus on in the coming months.

At this point I think its more productive to focus on doing this on an LMFDB-wide scale, rather than fixing individual missing friend links.  Of course if there is a separate purely arithmetic reason why two objects should be friends, we might want to record that as well, but L-function friends should be automatic (L-function friends are like family, you don't choose them, they just show up :)).

**Comment by davidfarmer** [comment]
Another (I think related) issue, is that the Galois conjugate of the L-function should also be
an L-function, presumably of another HMF.  But I can't seem to find it.
Maybe that HMF should give rise to two (Galois conjugate) L-functions?

**Comment by jvoight** [comment]
We can and should precompute the relationship between conjugate HMFs (in both senses, there is a Galois action on both the ideals and the coefficients, and these can have nontrivial interactions).  To do this in the most robust and interesting way, it would be analogous to (and include the analogue of) computing HMF inner twists.  To do this in a simple way is, as Drew mentions, is solved by comparing L-functions and that will happen in v1.2.

**Comment by jvoight** [comment]
Also see #2262

**Comment by JohnCremona** [comment]
@jvoight Surely there is nothing to do to compute conjugate HMFs which are conjugate in the sense that their coefficients are Galois conjugates?  We only store Galois orbits anyway (e.g. https://www.lmfdb.org/ModularForm/GL2/TotallyReal/3.3.1101.1/holomorphic/3.3.1101.1-73.3-b which has dimension 2 with coefficients in Q(sqrt(5)) is a Galois orbit of two Hilbert newforms.).    There will be two different L-functions attached to this orbit, and in general an HMF of dimensions d will have d L-functions, each of degree d*(2n) where the base field has degree n.  Comparing with a CMF example, e.g. https://www.lmfdb.org/ModularForm/GL2/Q/holomorphic/5168/2/a/l/, I see that we include both two degree 2 L-functions, and their product which has degree 4.

**Comment by jvoight** [comment]
@JohnCremona Yes, there is nothing to do for the "coefficient field conjugates".  This issue would concern itself with the interaction between those coefficient-conjugates and either twists or the Galois action on the base field.  This is something that happens purely in terms of modular forms, though of course it has implications for L-functions.  My proposal is that (eventually!) we do this like for CMFs, with the added flavor coming from Galois action on the base field.

**Comment by JohnCremona** [comment]
Agreed!
